classes/image_types_ota.bbclass

```diff
@@ -7,7 +7,11 @@
 # boot scripts, kernel and initramfs images
 #
 
-do_image_otaimg[depends] += "e2fsprogs-native:do_populate_sysroot \
+do_image_ota_ext4[depends] += "e2fsprogs-native:do_populate_sysroot \
+			${@'grub:do_populate_sysroot' if d.getVar('OSTREE_BOOTLOADER', True) == 'grub' else ''} \
+			${@'virtual/bootloader:do_deploy' if d.getVar('OSTREE_BOOTLOADER', True) == 'u-boot' else ''}"
+
+do_image_ota_tar[depends] += "e2fsprogs-native:do_populate_sysroot \
 			${@'grub:do_populate_sysroot' if d.getVar('OSTREE_BOOTLOADER', True) == 'grub' else ''} \
 			${@'virtual/bootloader:do_deploy' if d.getVar('OSTREE_BOOTLOADER', True) == 'u-boot' else ''}"
 
```

Comment:
Shouldn't grub/u-boot be a dependency of the do_otasetup task?

Also, for do_image_ota_tar we probably do not need e2fsprogs... So probably no direct dependencies needed for do_image_ota_tar.